### PR TITLE
beginning of adapting simulator to work with heterogenous cloud

### PR DESCRIPTION
**Pull request by pkoniu**

### CODE REVIEW

**Comment by pkoniu** [comment]
DAGStats probably need to be refactored due to the fact that we now have whole set of vmTypes not only one. We need to discuss whether or not the critical path for dag should be calculated for both dag and vmtype.

File: src/cws/core/simulation/Simulation.java [hunk 1]
```diff
@@ -322,13 +321,14 @@ public void runTest(CommandLine args) {
                 }
             }
 
-            DAGStats dagStats = new DAGStats(dag, environment.getVMType());
+            VMType vmTypeForDagStats = environment.getVmTypes().iterator().next();
```
Comment (aciepiela) [inline src/cws/core/simulation/Simulation.java]: We should select a best VM here as discussed: https://github.com/malawski/cloudworkflowsimulator/issues/168

Please add a TODO mark here so we remember to modify it after we decide on a solution for the issue (same as with `DAGDynamicScheduler`)

File: src/cws/core/algorithms/DynamicAlgorithm.java [hunk 2]
```diff
@@ -67,21 +68,22 @@ private int estimateVMsNumber() {
             return 0;
         }
 
-        return (int) Math.ceil(getMaxSpendingSpeedWeCanAfford() / getEnvironment().getSingleVMPrice());
+        VMType vmType = getEnvironment().getVmTypes().iterator().next();
```
Comment (aciepiela) [inline src/cws/core/algorithms/DynamicAlgorithm.java]: Same as in the comment below when it comes to choosing the best VM.

File: src/cws/core/simulation/Simulation.java [hunk 3]
```diff
@@ -231,7 +229,8 @@ public void runTest(CommandLine args) {
         double maxScaling = Double.parseDouble(args.getOptionValue("max-scaling", DEFAULT_MAX_SCALING));
         double alpha = Double.parseDouble(args.getOptionValue("alpha", DEFAULT_ALPHA));
 
-        VMType vmType = vmTypeLoader.determineVMTypes(args).iterator().next();
+        Set<VMType> vmTypes = vmTypeLoader.determineVMTypes(args);
+        VMType vmType = vmTypes.iterator().next();
         logVMType(vmType);
```
Comment (aciepiela) [inline src/cws/core/simulation/Simulation.java]: We could log all available vmTypes here

**Pull request merged**